### PR TITLE
remove componentWillMount from SecureAdminConsole

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- remove deprecated lifecycle method from `SecureAdminConsole`

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/52399/upgrade-kots-to-react-18

#### Special notes for your reviewer:
- none

## Steps to reproduce
load the app 

#### Does this PR introduce a user-facing change?
none- this is a refactor 

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/components/SecureAdminConsole.jsx [hunk 1]
```diff
@@ -207,7 +207,16 @@ class SecureAdminConsole extends React.Component {
     }
   }
 
-  async componentWillMount() {
```
Comment (rianfowler) [inline web/src/components/SecureAdminConsole.jsx:210]: Diff looks weird but I moved the logic from `componentWillMount` into `componentDidMount` only changing what is noted

File: web/src/components/shared/SubNavBar/SubNavBar.jsx [hunk 2]
```diff
@@ -28,7 +28,10 @@ export default function SubNavBar({
   // config view always shows the deployed version, falling back to the top version if nothing is deployed
   if (app?.downstream?.pendingVersions?.length) {
     kotsSequence = app?.downstream?.pendingVersions[0]?.parentSequence;
-    if (!app?.downstream?.currentVersion || app?.downstream?.gitops?.isConnected) {
```
Comment (rianfowler) [inline web/src/components/shared/SubNavBar/SubNavBar.jsx:31]: unrelated linting change

File: web/src/components/snapshots/SnapshotDetails.jsx [hunk 3]
```diff
@@ -86,23 +86,23 @@ class SnapshotDetails extends Component {
...
             w.globals.labels[dataPointIndex] +
             "</p>" +
-            "<span class=\"u-fontSize--small u-fontWeight--normal u-textColor--bodyCopy u-marginTop--10\">" +
+            '<span class="u-fontSize--small u-fontWeight--normal u-textColor--bodyCopy u-marginTop--10">' +
             w.globals.seriesZ[seriesIndex][dataPointIndex] +
             "</span>" +
             "<br />" +
             "<br />" +
-            "<span class=\"u-fontSize--normal u-fontWeight--normal u-textColor--bodyCopy u-marginTop--10\">" +
+            '<span class="u-fontSize--normal u-fontWeight--normal u-textColor--bodyCopy u-marginTop--10">' +
```
Comment (rianfowler) [inline web/src/components/snapshots/SnapshotDetails.jsx:98]: unrelated linting change

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.jsx [hunk 4]
```diff
@@ -401,7 +401,10 @@ class AppVersionHistoryRow extends Component {
             <ReactTooltip effect="solid" className="replicated-tooltip" />
           </div>
         )}
-        {(isPastVersion || isCurrentVersion || isPendingDeployedVersion || version?.status === "superseded") &&
+        {(isPastVersion ||
```
Comment (rianfowler) [inline web/src/features/AppVersionHistory/AppVersionHistoryRow.jsx:404]: unrelated linting change

File: web/src/components/SecureAdminConsole.jsx [hunk 5]
```diff
@@ -216,12 +225,12 @@ class SecureAdminConsole extends React.Component {
         token: token.replace(/"/g, ""),
         sessionRoles: Utilities.getCookie("session_roles"),
       };
-      const loggedIn = await this.completeLogin(loginData);
-      if (loggedIn) {
-        Utilities.removeCookie("token");
-        Utilities.removeCookie("session_roles");
-      }
-      return;
+      const loggedIn = this.completeLogin(loginData).then(() => {
```
Comment (rianfowler) [inline web/src/components/SecureAdminConsole.jsx:228]: `componentWillMount` had an async function in it. I didn't do that for `componentDidMount` - I replaced the `await` with `.then`

Comment (alicenstar) [inline web/src/components/SecureAdminConsole.jsx:228]: How is this working? it looks like it's trying to access a variable (`loggedIn`) in the outer scope.

Comment (rianfowler) [reply web/src/components/SecureAdminConsole.jsx]: whoops

**Review by alicenstar — APPROVED** [review]
